session.go

```diff
@@ -51,6 +51,15 @@ type Session interface {
 	// Warnings returns a list of warnings generated for the session on connect/auth
 	Warnings() []error
 
+	// ListenAndForward creates a new Tunnel which will listen for new inbound
+	// connections. Connections on this tunnel are automatically forwarded to
+	// the provided URL.
+	ListenAndForward(ctx context.Context, backend *url.URL, cfg config.Tunnel) (Forwarder, error)
+
+	// ListenAndServeHTTP creates a new Tunnel to serve as a backend for an HTTP server. Connections will be
+	// forwarded to the provided HTTP server.
+	ListenAndServeHTTP(ctx context.Context, cfg config.Tunnel, server *http.Server) (Tunnel, error)
```

Comment:
This should probably return a Forwarder as well, so as not to expose the Tunnel's Accept call and cause weirdness with the HTTP server.